### PR TITLE
OCM-7498 | test: automate id:OCP-72715 create/edit nodepool with node…

### PR DESCRIPTION
**Pull request by aaraj7**
[OCM-7498](https://issues.redhat.com//browse/OCM-7498) | test: automate id:OCP-72715 create/edit nodepool with node_drain_grace_period to HCP cluster via ROSA cli can work well

output log :- https://privatebin.corp.redhat.com/?1ad49d4e676ed805#8v53avRQLUaTRr6uunhpBqV9fUjhSbgD5qQtEeNxzmCC

### CODE REVIEW

File: tests/e2e/test_rosacli_node_pool.go [hunk 1]
```diff
@@ -730,4 +730,78 @@ var _ = Describe("Edit nodepool",
 				analyzeUpgrade(nodePoolManualName, "manual")
 				analyzeUpgrade(nodePoolAutoName, "automatic")
 			})
+
+		It("create/edit nodepool with node_drain_grace_period to HCP cluster via ROSA cli can work well - [id:72715]",
+			labels.Medium, labels.NonClassicCluster,
```
Comment (yingzhanredhat) [inline tests/e2e/test_rosacli_node_pool.go:735]: The priority is high

Comment (aaraj7) [reply tests/e2e/test_rosacli_node_pool.go]: oh yes, sorry, changed it

File: tests/e2e/test_rosacli_node_pool.go [hunk 2]
```diff
@@ -730,4 +730,78 @@ var _ = Describe("Edit nodepool",
...
+			func() {
+				By("check help message for create/edit machinepool")
+				help, err := machinePoolService.RetrieveHelpForCreate()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(help.String()).To(ContainSubstring("--node-drain-grace-period"))
+				help, err = machinePoolService.RetrieveHelpForEdit()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(help.String()).To(ContainSubstring("--node-drain-grace-period"))
+
+				nodeDrainGracePeriodsReqAndRes := []map[string]string{{"20": "20 minutes", "20 hours": "1200 minutes", "20 minutes": "20 minutes"}}
```
Comment (yuwang-RH) [inline tests/e2e/test_rosacli_node_pool.go:745]: Here needs a step with clear description,rather bellow By("check help message for create/edit machinepool") as below code is not testing the help message

File: tests/e2e/test_rosacli_node_pool.go [hunk 3]
```diff
@@ -730,4 +730,78 @@ var _ = Describe("Edit nodepool",
...
+				By("Describe nodepool")
+				output, err := machinePoolService.DescribeAndReflectNodePool(clusterID, nodePoolName)
+				Expect(err).ToNot(HaveOccurred())
+				if nodeDrainGracePeriodForCluster == "0" {
+					Expect(output.NodeDrainGracePeriod).To(Equal(""))
+				} else {
+					Expect(output.NodeDrainGracePeriod).To(Equal(nodeDrainGracePeriodForCluster))
+				}
+
+				nodeDrainGracePeriodsReqAndRes = []map[string]string{{"10": "10 minutes", "10 hours": "600 minutes", "10 minutes": "10 minutes"}}
```
Comment (yuwang-RH) [inline tests/e2e/test_rosacli_node_pool.go:789]: A step with clear description is needs for below testing code.

File: tests/e2e/test_rosacli_node_pool.go [hunk 4]
```diff
@@ -730,4 +730,78 @@ var _ = Describe("Edit nodepool",
...
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Describe nodepool")
+						output, err := machinePoolService.DescribeAndReflectNodePool(clusterID, nodePoolName)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(output.NodeDrainGracePeriod).To(Equal(res))
+					}
+				}
+			})
```
Comment (yuwang-RH) [inline tests/e2e/test_rosacli_node_pool.go:806]: We'd better to do clean-up step after the testing, in this case, to delete the created machinepools in this case

Comment (aaraj7) [reply tests/e2e/test_rosacli_node_pool.go:808]: It is being cleaned through afterEach() function inside describe block, so NO need to duplicate it I guess, wdyt?

Comment (yuwang-RH) [reply tests/e2e/test_rosacli_node_pool.go:808]: It is fine if there is clean-up step in afterEach block.

Comment (aaraj7) [reply tests/e2e/test_rosacli_node_pool.go:808]: Thanks

**Pull request merged**